generators/generators-commons/src/main/java/uk/gov/justice/services/generators/commons/helper/MessagingResourceUri.java

```diff
@@ -26,7 +26,16 @@ public String destinationName() {
      * @return capitalised uri string without delimiters
      */
     public String toCapitalisedString() {
-        return capitalize(stripStart(uriString, "/"), '.').replace(".", "");
+        //Remove any character that is invalid as starting character for a class name
+        String uriStringStriped = uriString.replaceAll("^[^A-Za-z_$]+", "");
+
+        final StringBuilder nonWordCharAsString = new StringBuilder();
+        uriString.chars().filter(e -> !Character.isLetter(e)).forEach(e -> nonWordCharAsString.append((char) e));
+        char[] delimiters = nonWordCharAsString.toString().toCharArray();
+        String capitalizedWithCharacter = capitalize(uriStringStriped, delimiters);
```

Comment:
This is a very round-the-houses bit of code. I suggest simplifying as follows:
cleaned = orig.replaceAll("[^A-Za-z0-9_$]", " ")  - ie, replace all other chars with a space
captialised = capitaliseFully(cleaned, {' ', '_', '$'}) - you have a fixed set of delimiters now so this is simple
result = capitalised.replaceAll(" ", "") - remove the spaces
then remove the 1st char if it is a number, possibly in a separate method
Much of this could be inline to avoid variable names for partial results that don't really mean anything